### PR TITLE
CADC-12765 merged in the cavern uid/gid PR

### PR DESCRIPTION
**Pull request by jburke-cadc**
Merged in the uid/gid PR, and updated LocalServiceURI to use cavern.properties.

### CODE REVIEW

**Review by pdowler — CHANGES_REQUESTED** [review]
now that I look at this change to get rid of that config file, this seems like a bad idea... I don't know if someone else (INAF maybe) still builds and runs a vospace based on our libs.

What about keep the old constants and add ALT_CONFIG_FILE and ALT_RESOURCE_ID_KEY? Then look for both files and use the one that is found with the associated key?

**Comment by jburke-cadc** [comment]
That seems like a much better idea. Making a minor version specifically for this use case doesn't stop others from unknowingly downloading it and breaking their app. I'll update.

**Comment by pdowler** [comment]
could also then be 1.3.5

File: cavern/src/main/java/org/opencadc/cavern/CavernInitAction.java [hunk 1]
```diff
@@ -86,6 +91,21 @@ public CavernInitAction() {
...
     public void doInit() {
         try {
+            // Check config file
+            CavernConfig cavernConfig = new CavernConfig();
+            MultiValuedProperties props = cavernConfig.getConfig();
+
+            // create root directories for node/files
+            String baseDir = props.getFirstPropertyValue(CavernConfig.FILESYSTEM_BASE_DIR);
+            String subPath = props.getFirstPropertyValue(CavernConfig.FILESYSTEM_SUB_PATH);
+            Path root = Paths.get(baseDir + subPath);
```
Comment (pdowler) [inline cavern/src/main/java/org/opencadc/cavern/CavernInitAction.java:101]: make sure there is a `/` between these?

File: cavern/src/main/java/org/opencadc/cavern/FileSystemNodePersistence.java [hunk 2]
```diff
@@ -111,23 +109,18 @@ public class FileSystemNodePersistence implements NodePersistence {
...
     public FileSystemNodePersistence() {
         CavernConfig cavernConfig = new CavernConfig();
         this.config = cavernConfig.getConfig();
-        String rootConfig = config.getFirstPropertyValue(CavernConfig.FILESYSTEM_BASE_DIR);
-        if (rootConfig == null) {
-            throw new RuntimeException("CONFIG: Failed to find " + CavernConfig.FILESYSTEM_BASE_DIR);
-        }
-        this.root = Paths.get(rootConfig);
-        
+        String baseDir = config.getFirstPropertyValue(CavernConfig.FILESYSTEM_BASE_DIR);
```
Comment (pdowler) [inline cavern/src/main/java/org/opencadc/cavern/FileSystemNodePersistence.java:117]: Why repeat this code? Why not `this.root = config.getNodeRoot();`??

File: cavern/src/main/java/org/opencadc/cavern/FileSystemNodePersistence.java [hunk 3]
```diff
@@ -111,23 +109,18 @@ public class FileSystemNodePersistence implements NodePersistence {
...
+        String subPath = config.getFirstPropertyValue(CavernConfig.FILESYSTEM_SUB_PATH);
+        this.root = Paths.get(baseDir + subPath);
+
         // must be hard coded to this and not set via java system properties
         this.identityManager = new PosixIdentityManager();
-        
+
         LocalAuthority loc = new LocalAuthority();
         // TODO: move constant to cadc-registry
         URI posixMapperID = loc.getServiceURI("http://www.opencadc.org/std/posix#group-mapping-1.0");
```
Comment (pdowler) [inline cavern/src/main/java/org/opencadc/cavern/FileSystemNodePersistence.java:126]: I have fixed this in my current code to:
```
URI posixMapperID = loc.getServiceURI(Standards.POSIX_GROUPMAP.toASCIIString());
```
but not pushed yet.

File: cavern/src/test/resources/cadc-registry.properties [hunk 4]
```diff
@@ -8,4 +8,4 @@ ca.nrc.cadc.reg.client.RegistryClient.baseURL = https://example.net/reg
 ivo://ivoa.net/std/GMS#search-1.0 = ivo://cadc.nrc.ca/gms
 ivo://ivoa.net/std/UMS#users-0.1 = ivo://cadc.nrc.ca/gms
 
-http://www.opencadc.org/std/posix#group-mapping-1.0 = ivo://cadc.nrc.ca/gms
\ No newline at end of file
+http://www.opencadc.org/std/posix#group-mapping-1.0 = ivo://cadc.nrc.ca/gms
```
Comment (pdowler) [inline cavern/src/test/resources/cadc-registry.properties:11]: again, 0.1

**Review by pdowler — CHANGES_REQUESTED** [review]

File: cavern/src/main/java/org/opencadc/cavern/CavernInitAction.java (hunk 1 above)
Comment (jburke-cadc) [reply cavern/src/main/java/org/opencadc/cavern/CavernInitAction.java:101]: Yes, would be a good idea if the base doesn't end with a / and the subPath doesn't start with one.

File: cavern/src/main/java/org/opencadc/cavern/FileSystemNodePersistence.java (hunk 3 above)
Comment (jburke-cadc) [reply cavern/src/main/java/org/opencadc/cavern/FileSystemNodePersistence.java:126]: Updated my copy.

File: cavern/src/test/resources/cadc-registry.properties (hunk 4 above)
Comment (jburke-cadc) [reply cavern/src/test/resources/cadc-registry.properties:11]: Updated.

**Review by pdowler — APPROVED** [review]

**Pull request merged**